src/firebase/AuthService.ts

```diff
@@ -1,31 +1,39 @@
 import firebase from "firebase/app";
 import "firebase/auth";
 
-class AuthService {
-    provider: firebase.auth.OAuthProvider;
-    accessToken: string | undefined;
-    idToken: string | undefined;
+interface ProfileData {
+    isAdmin: boolean;
+    isSnapper: boolean;
+}
 
-    constructor() {
-        this.provider = new firebase.auth.OAuthProvider("microsoft.com");
-        this.provider.setCustomParameters({
-            prompt: process.env.REACT_APP_PROMPT,
-            // Only users from a particular Azure AD tenant to sign into the application
-            tenant: process.env.REACT_APP_TENANT,
-        });
+export async function getOrCreateUserProfile(): Promise<ProfileData> {
```

Comment:
The contents of this function should be refactored into distinct, descriptively named functions that encapsulate lower-level functionality.

Something similar to (but not prescriptively) this:
```
assertUserIsAuthenticated()
profile = getUserProfile()
if (!profileExists(profile)) {
  profile = createProfile();
}
return profile;
```